File: Backend/src/main/java/com/app/logistica/services/DriverService.java
package com.app.logistica.services;

import com.app.logistica.dtos.driver.DriverRequest;

import java.util.List;

public interface DriverService {
    DriverRequest createDriver(DriverRequest driverRequest);
    DriverRequest updateDriver(Long driverID, DriverRequest driverRequest);
    String deleteDriver(Long driverID);
    DriverRequest getDriver(Long driverID);
    List<DriverRequest> getDrivers();

}

File: ReactWeb/src/services/DriverService.js
import api from "./api.js"

export const getAllDrivers = () => api.get("/drivers");
export const getDriverById = (id) => api.get(`/drivers/${id}`);
export const createDriver = (driver) => api.post("/drivers", driver);
export const updateDriver = (id, driver) => api.put(`/drivers/${id}`, driver);
export const deleteDriver = (id) => api.delete(`/drivers/${id}`);
